File: Cpp/The_Two_Sneaky_Numbers_of_Digitville.cpp
class Solution {
public:
    vector<int> getSneakyNumbers(vector<int>& nums) {
        int n = nums.size() - 2;
        int xorAll = 0;
        for (int i = 0; i < nums.size(); i++) {
            xorAll ^= nums[i];
        }
        for (int i = 0; i < n; i++) {
            xorAll ^= i;
        }
        int setBit = xorAll & -xorAll;
        int x = 0, y = 0;

        for (int num : nums) {
            if (num & setBit)
                x ^= num;
            else
                y ^= num;
        }

        for (int i = 0; i < n; i++) {
            if (i & setBit)
                x ^= i;
            else
                y ^= i;
        }

        return {x, y};
    }
};

File: Rust/The_Two_Sneaky_Numbers_of_Digitville.rs
impl Solution {
    pub fn get_sneaky_numbers(nums: Vec<i32>) -> Vec<i32> {
        let n = nums.len() - 2;
        let mut xor_all = 0;
        for &num in &nums {
            xor_all ^= num;
        }
        for i in 0..n {
            xor_all ^= i as i32;
        }
        let set_bit = xor_all & -xor_all;
        let mut x = 0;
        let mut y = 0;

        for &num in &nums {
            if num & set_bit != 0 {
                x ^= num;
            } else {
                y ^= num;
            }
        }

        for i in 0..n {
            let i32_val = i as i32;
            if i32_val & set_bit != 0 {
                x ^= i32_val;
            } else {
                y ^= i32_val;
            }
        }

        vec![x, y]
    }
}
